### ISSUE TITLE
a11y: nav logo is not accessible if there is no siteTitle

### ISSUE THREAD

**Issue by gorlanova** [vuejs/vitepress#3277]
### Describe the bug

When displaying a logo in the navbar, but setting the siteTitle to `false`, it triggers an accessibility problem : the logo link has no accessible label

<img width="344" alt="Screenshot 2023-11-30 at 12 13 35" src="https://github.com/vuejs/vitepress/assets/58045985/454fad23-4c1d-49eb-a037-c688073d9c71">

### Reproduction

- set a logo for your website in themeConfig
- set siteTitle to `false`
- check the accessible label of the logo link in the navbar: there is none

### Expected behavior

Links should have an accessible label, whether this is set manually or automatically generated (from the defineConfig title ? or maybe siteTitle could be defined but we could have another prop to "hide" it visually ?)

### System Info

```Text
System:
    OS: macOS 14.1.1
    CPU: (8) arm64 Apple M2
    Memory: 232.50 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.3.1 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.7 - /opt/homebrew/bin/npm
    pnpm: 8.9.2 - /opt/homebrew/bin/pnpm
  Browsers:
    Safari: 17.1
  npmPackages:
    vitepress: ^1.0.0-rc.31 => 1.0.0-rc.31
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
You need to set the alt:

```
logo: { src: '...', alt: '...' }
```
